### DISCUSSION TITLE
Porting Doom

### DISCUSSION THREAD

**Discussion by shoaib-jamal** [prochazkaml/MichalOS#8] [Ideas]
Doom is probably the most famous game and every hobby project is now ft. it.
KolibriOS & MenuetOS are both AssemblyOS and they have sucessfully ported doom and quake in their OS
MichalOS would rock if it had a 3D library

**Comment by prochazkaml** [comment]
Yes, KolibriOS and MenuetOS are operating systems which were written in assembly, however, these are 32-bit operating systems, which have access to ~3 GB of RAM. Meanwhile, MichalOS is a 16-bit OS, so the best you can get is 640 kB of RAM. Looking at the minimum requirements of Doom (4 MB of RAM), there's no way this could work. And no, I can't just easily make MichalOS into a 32-bit OS, that would be near impossible, because I would lose access to BIOS calls, which are fundamental for MichalOS to operate.

**Reply by shoaib-jamal** [reply]
Yes you are right... So there would be no 3d Library for MichalOS in future?

**Reply by shoaib-jamal** [reply]
Maybe see this project https://github.com/nanochess/cubicDoom[https://github.com/nanochess/cubicDoom] its also a recreation of doom and is a bootsector game.

**Reply by shoaib-jamal** [reply]
Ignore the [ ] brackets

**Reply by prochazkaml** [reply]
> Yes you are right... So there would be no 3d Library for MichalOS in future?

Probably not.

> Maybe see this project [https://github.com/nanochess/cubicDoom[https://github.com/nanochess/cubicDoom]](https://github.com/nanochess/cubicDoom%5Bhttps://github.com/nanochess/cubicDoom%5D) its also a recreation of doom and is a bootsector game.

That look very interesting. It's really low-res and the performance isn't great, but still, it's quite the achievement of fitting it in 512 bytes. I could probably write a simple 3D raycaster and create a Wolfenstein 3D-like engine, but that would take some time.

**Reply by shoaib-jamal** [reply]
Yes I know it's a hard work and this project should be under heavy development to make it more popular. Best of Luck!
